✨ [amp-google-read-aloud-player] Adds the `amp-google-read-aloud-player` extension (#37402)
ampproject/amphtml: extensions/amp-google-read-aloud-player/amp-google-read-aloud-player.md

The intro is played whenever a listener starts playback from the beginning.</td>
  </tr>
  <tr>
    <td width="20%"><strong>data-outro</strong></td>
    <td>The URL of the MP3 audio file which will be played after the player ends reading. The 
outro can be used by publishers to create a unique ending for each playback.</td>
  </tr>
  <tr>
    <td width="20%"><strong>data-ad-tag-url</strong></td>
    <td>A complete <a href ="https://support.google.com/admanager/table/9749596">VAST ad tag URL</a> 
with parameters for playing pre-roll audio ads. When present, audio ads will be played before 
reading aloud page contents, after the user clicks on the play button for the first time. If not 
present, only page content will be played.<br/>
The ad tag URL must include the following parameters:

<ul>
  <li>iu=[Ad unit code]</li>
  <li>sz=1x1</li>
  <li>ad_type=audio</li>
  <li>output=vast</li>
  <li>unviewed_position_start=1</li>
  <li>env=instream</li>
  <li>vad_type=linear</li>
</ul></td>
  </tr>
</table>

## Usage

Include an `amp-google-read-aloud-player` on your AMP document:

```html
<amp-google-read-aloud-player

>   layout="fixed-height"

Reviewer: Same comment about unnecessary attribute. Apologies for not catching it on the previous review.
```suggestion
```

---
$category@: media
formats:
  - websites
teaser:
  text: Embeds the Google Read Aloud Player.
---

<!--
Copyright 2019 The AMP HTML Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS-IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->

# amp-google-read-aloud-player

## Behavior

Embeds the Google Read Aloud web player which enables reading out loud web pages.

> ## Supported Layouts
> 
> `fixed` and `fixed-height`.

Reviewer: Supported layouts are added to documentation automatically. Remove section.
```suggestion
```

specific API key.
<br><br>
When the parameter isn’t provided, the player uses the Canonical URL link tag of the current page.
This is useful for reading aloud the content of the current page and only a single Read Aloud
player is embedded in the page (no infinite scroll).
<br><br>
This value overrides it, and publishers are required to set it only in these cases:
<ol>
  <li>Other page - When the player is embedded in a different page than the one that its content
should be read aloud. In this case, set this parameter to the canonical URL of the other page.</li>
  <li>Multiple players - When there are multiple players on the same page, see Multiple players
(infinite scroll). In this case, set this parameter to the canonical URL of the article that the
player should read aloud.</li>
  <li>Canonical URL tag is missing - When the Canonical URL link tag of the current page is missing
(and cannot be added). In this case, set this parameter to the canonical URL of the page.</td>
</li>
</ol>
  </tr>
  <tr>
    <td width="20%"><strong>data-speakable</strong></td>
    <td>Controls the contents to read aloud.
<br><br>

> Possible values:
> <ul>
>   <li>false - Google automatically analyzes the web page and decides which text parts are suitable
> for reading aloud.</li>
>   <li>true - Google uses only text from elements marked with the speakable markup on the web page.
> </li></td>

Reviewer: `true` and `false` are confusing since they're not string values. You could reword this to refer to attribute presence instead.

I would also add a documentation link that explains "speakable markup".

```suggestion
When present, uses text only from elements annotated with [speakable markup](https://...). Otherwise, Google analyzes the document automatically to choose the suitable text parts.
```

The intro is played whenever a listener starts playback from the beginning.</td>
  </tr>
  <tr>
    <td width="20%"><strong>data-outro</strong></td>
    <td>The URL of the MP3 audio file which will be played after the player ends reading. The 
outro can be used by publishers to create a unique ending for each playback.</td>
  </tr>
  <tr>
    <td width="20%"><strong>data-ad-tag-url</strong></td>
    <td>A complete <a href ="https://support.google.com/admanager/table/9749596?hl=en">VAST ad tag URL</a> 
with parameters for playing pre-roll audio ads. When present, audio ads will be played before 
reading aloud page contents, after the user clicks on the play button for the first time. If not 
present, only page content will be played.<br/>
The ad tag URL must include the following parameters:

<ul>
  <li>iu=[Ad unit code]</li>
  <li>sz=1x1</li>
  <li>ad_type=audio</li>
  <li>output=vast</li>
  <li>unviewed_position_start=1</li>
  <li>env=instream</li>
  <li>vad_type=linear</li>
</ul></td>
  </tr>
</table>

## Usage

Include an `amp-google-read-aloud-player` on your AMP document:

```html

> <amp-google-read-aloud-player
>   layout="fixed-height" height="65"
>   data-api-key=<YOUR_API_KEY>
>   data-tracking-ids=<YOUR_TRACKING_IDS>
>   data-voice=<YOUR_VOICE>>
> </amp-google-read-aloud-player>

Reviewer: nit: formatting
```suggestion
<amp-google-read-aloud-player
  layout="fixed-height"
  height="65"
  data-api-key="<YOUR_API_KEY>"
  data-tracking-ids="<YOUR_TRACKING_IDS>"
  data-voice="<YOUR_VOICE>"
></amp-google-read-aloud-player>
```
